packages/obojobo-express/routes/api/media.js

```diff
@@ -0,0 +1,86 @@
+const express = require('express')
+const router = express.Router()
+const multer = require('multer')
+
+const mediaConfig = oboRequire('config').media
+const MediaModel = oboRequire('models/media')
+
+const diskStorage = multer.diskStorage({
+	destination: mediaConfig.tempUploadDestination
+})
+
+// single() takes the field name of the file field
+//  in the multipart form being submitted
+const multerUpload = multer({
+	storage: diskStorage,
+	limits: {
+		fileSize: mediaConfig.maxUploadSize
+	},
+	fileFilter: (req, file, cb) => {
+		const isValidFile = MediaModel.isValidFileType(file.originalname, file.mimetype)
+
+		if (isValidFile) {
+			// there is no error, accept the file
+			return cb(null, true)
+		}
+
+		// there is an error, do not accept the file
+		cb(
+			'File upload only supports the following filetypes: ' +
+				mediaConfig.allowedMimeTypesRegex.split('|').join(', '),
+			false
+		)
```

Comment:
Could you reverse these so the error state returns first and then the non-error state happens after that?

Reply:
Fixed in commit f6b318266d7492dd1711f8321f5df5c10962d521